### PR TITLE
Fix #420 about toHaveLength()

### LINKED ISSUE #420: Bug: toHaveLength() does not work with accented letters and multi-byte strings
These tests fail with Kahlan:
```php
...
        it("passes with accented letters", function () {
            expect('Olá')->toHaveLength(3);
        });

        it("passes with multibyte characters, like emojis", function () {
            expect('Hello 😉')->toHaveLength(7);
        });
```
I'll submit a PR with the fix.

### PR DESCRIPTION
**Pull request by thiagodp**
See #420

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -9,7 +9,8 @@
         }
     ],
     "require": {
-        "php": ">=7.2"
+        "php": ">=7.2",
+        "ext-mbstring": "*"
```
[the remark this answers is not in the document]
Comment (thiagodp) [reply composer.json:13]: Hello @samsonasik,
problem is that users do not expect that `toHaveLength()` will fail with such valid strings.
Kahlan support to multi-byte strings is essential for modern applications.

[the remark this answers is not in the document]
Comment (jails) [reply composer.json:13]: Well, I'm agree with @samsonasik on this one. Multi-byte strings are an interpretation of a series of bytes. There's a lot of different encoding systems and even if UTF-8 is a de facto standard, it doesn't change the bytes length of a string. Javascript is also behaving the same way on strings length definition:
Screenshot_20230924_230115

Imo `toHaveLength()` must return the number of bytes of a string. By the way Jest's `toHaveLength()` matcher also behave the same way as Kahlan (https://github.com/jestjs/jest/blob/a640c6513f08f6d97827050010a8a28fff530ac9/packages/expect/src/matchers.ts#L674)

If you deal with multi-byte strings imo it should be better to create some custom matchers or simply do like @samsonasik explained:

```php
expect(mb_strlen('Hello 😉'))->toBe(7);
```

**Comment by thiagodp** [comment]
Hello @jails , I respectfully disagree. A matcher named `toHaveBytes()` would be appropriated for that purpose (how many bytes the string contains). A matcher named `toHaveLength()`  should return how many _characters_ the programmer expects, not how many _bytes_.

Kahlan's API could be better than the alternatives, instead of just being compatible. The API already has some differences and lacks some matchers.

**Comment by jails** [comment]
Assuming programmers except that the content of a variable is necessarily a string which may contain multi-byte characters is not what I'm expecting personnally. There's number of abstractions which doesn't make this assumption starting with Javascript itselft and also other describe/it test frameworks. 

So I would advice to create a custom `toHaveStringLength()` matcher (see https://kahlan.github.io/docs/matchers.html#custom) so it will match your own logic as a developper.

**Pull request closed without merging**